code/game/mecha/mecha_parts.dm

```diff
@@ -232,6 +232,13 @@
 	name = "\improper Phazon chassis"
 	construct_type = /datum/component/construction/unordered/mecha_chassis/phazon
 
+/obj/item/mecha_parts/chassis/phazon/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(!istype(I, /obj/item/assembly/signaler/anomaly/bluespace) && !istype(I, /obj/item/circuitboard/))
+		to_chat(user, "\The [I.name] doesn't fit in the anomaly core socket. Make sure you're using a Bluespace core.")

```

Comment:
This is going to throw the error every time you use any item on the chassis that isn't a bluespace core or circuit board, right? It should probably be checking for whatever variable the chassis uses to hold it's current state; batteries, capacitors, wires, and the various tools required to build a mech are all not bluespace cores, but they are still necessary.

Alternatively, you could do a check for anomaly cores that aren't bluespace;
`if(istype(I, /obj/item/assembly/signaler/anomaly) && !istype(I, /obj/item/assembly/signaler/anomaly/bluespace))`

Also, you don't need the `else return` part at all.

Reply:
That check is what I was looking for. I wasn't certain how to syntax it so it would do that. I'll make that change and test it later on tonight.